server/aws-lsp-codewhisperer/src/language-server/chat/chatSessionManagementService.ts

```diff
@@ -0,0 +1,45 @@
+import { CredentialsProvider } from '@aws/language-server-runtimes/server-interface'
+import { ChatSessionService, ChatSessionServiceConfig } from './chatSessionService'
+
+export class ChatSessionManagementService {
+    #sessionByTab: Map<string, ChatSessionService>
+    #credentialsProvider: CredentialsProvider
+    #clientConfig?: ChatSessionServiceConfig | (() => ChatSessionServiceConfig)
+
+    constructor(
+        credentialsProvider: CredentialsProvider,
+        clientConfig?: ChatSessionServiceConfig | (() => ChatSessionServiceConfig)
+    ) {
+        this.#credentialsProvider = credentialsProvider
+        this.#sessionByTab = new Map<string, any>()
+        this.#clientConfig = clientConfig
+    }
+
+    public hasSession(tabId: string): boolean {
+        return this.#sessionByTab.has(tabId)
+    }
+
+    public getSession(tabId: string): ChatSessionService {
+        const maybeSession = this.#sessionByTab.get(tabId)
+
+        if (!maybeSession) {
+            const clientConfig = typeof this.#clientConfig === 'function' ? this.#clientConfig() : this.#clientConfig
+            const newSession = new ChatSessionService(this.#credentialsProvider, clientConfig)
```

Comment:
I am separating this part out into a createSession function in future work https://github.com/aws/language-servers/commit/1e0798053214ba66e4330ec1b6ff40f312f6c74b#diff-f6d1aac51c11d25d6c080c419e43328dee44efcd423b490c0f24b44453fcb841